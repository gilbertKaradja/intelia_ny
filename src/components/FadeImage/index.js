import React, { Component } from 'react';

class FadeImage extends Component {
    constructor() {
        super();

        this.state = {
            imageLoaded: false
        }

        this.onLoadHandler = this.onLoadHandler.bind(this);
    }

    componentDidUpdate(prevProps) {
        if(prevProps.src !== this.props.src) {
            this.setState({imageLoaded: false})
        }
    }

    render() {
        
        return (
            <div className="FadeImage">

                <img src={this.props.src} onLoad={this.onLoadHandler} />

                <div className={this.state.imageLoaded ? 'placeholder hidden' : 'placeholder'}></div>
            </div>
        );
    }

    onLoadHandler() {
        this.setState({ imageLoaded: true });
    }
}

export default FadeImage;