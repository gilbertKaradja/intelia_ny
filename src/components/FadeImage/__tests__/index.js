import React from 'react';
import { shallow, mount } from 'enzyme';

import FadeImage from '../index.js';

import testImage from './test_image.jpg';
import testImage2 from './test_image_2.png';

describe('FadeImage component', () => {

    it('should render without issues', () => {

        //renders initial dropdown properly
        const component = shallow(
            <FadeImage
                src={testImage}
            />
        );
        expect(component).toMatchSnapshot();

        component.unmount();
    });

    it('should call onLoad function', () => {

        const spy = jest.spyOn(FadeImage.prototype, 'onLoadHandler');

        const component = mount(
            <FadeImage
                src={testImage}
            />
        );

        component.instance().onLoadHandler();
        expect(spy).toHaveBeenCalled();

        component.unmount();
    });

    it('should swap images when src is changed', () => {
        
        const component = shallow(
            <FadeImage
                src={testImage}
            />
        );
        expect(component).toMatchSnapshot();

        component.setProps({
            src: testImage2
        });

        expect(component).toMatchSnapshot();
    });
});