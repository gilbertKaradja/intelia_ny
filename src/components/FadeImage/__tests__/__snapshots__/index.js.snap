// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FadeImage component should render without issues 1`] = `
<div
  className="FadeImage"
>
  <img
    onLoad={[Function]}
    src="test_image.jpg"
  />
  <div
    className="placeholder"
  />
</div>
`;

exports[`FadeImage component should swap images when src is changed 1`] = `
<div
  className="FadeImage"
>
  <img
    onLoad={[Function]}
    src="test_image.jpg"
  />
  <div
    className="placeholder"
  />
</div>
`;

exports[`FadeImage component should swap images when src is changed 2`] = `
<div
  className="FadeImage"
>
  <img
    onLoad={[Function]}
    src="test_image_2.png"
  />
  <div
    className="placeholder"
  />
</div>
`;
